sssd_test_framework/utils/sss_override.py

```diff
@@ -0,0 +1,402 @@
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                # Create local override
+                # SSSD must be running for this to work
+                client.sssd.start()
+                client.sss_override.group("users").add(name="o-users", gid=10001)
+
+                # Assert the overridden name and gid
+                result = client.tools.getent.group("o-users")
+                assert result is not None
+                assert result.gid == 10001
+
+        :param group: Group name.
+        :type group: str
+        :return: New group object.
+        :rtype: SSSOverrideGroup
+        """
+        return SSSOverrideGroup(self, group)
+
+
+class SSSOverrideUser:
+    """
+    Management of local override for users using sss_override.
+    """
+
+    def __init__(self, util: SSSOverrideUtils, user: str) -> None:
+        """
+        :param util: OverrideUser object.
+        :type util: SSSOverrideUser
+        :param user: User name.
+        :type user: str
+        """
+        self.util: SSSOverrideUtils = util
+        self.user: str = user
+
+    def add(
+        self,
+        *,
+        uid: int | None = None,
+        gid: int | None = None,
+        name: str | None = None,
+        home: str | None = None,
+        gecos: str | None = None,
+        shell: str | None = None,
+        certificate: str | None = None,
+    ) -> SSSOverrideUser:
+        """
+        Create new local override for user.
+
+        :param uid: User id, defaults to None
+        :type uid: int | None, optional
+        :param gid: Primary group id, defaults to None
+        :type gid: int | None, optional
+        :param name: | None, optional
+        :type name: str | None, optional
+        :param home: Home directory, defaults to None
+        :type home: str | None, optional
+        :param gecos: GECOS, defaults to None
+        :type gecos: str | None, optional
+        :param shell: Login shell, defaults to None
+        :type shell: str | None, optional
+        :param certificate: PKI certificate defaults to None
+        :type certificate: str | None, optional
+        :return: Self.
+        :rtype: SSSOverrideUser
+        """
+        args: CLIBuilderArgs = {
+            "name": (self.util.cli.option.VALUE, name),
+            "uid": (self.util.cli.option.VALUE, uid),
+            "gid": (self.util.cli.option.VALUE, gid),
+            "home": (self.util.cli.option.VALUE, home),
+            "gecos": (self.util.cli.option.VALUE, gecos),
+            "shell": (self.util.cli.option.VALUE, shell),
+            "certificate": (self.util.cli.option.VALUE, certificate),
+        }
+
+        self.util.logger.info(f"Creating local override for user {self.user} on {self.util.host.hostname}")
+        self.util.host.ssh.exec(["sss_override", "user-add", self.user] + self.util.cli.args(args))
+
+        return self
+
+    def modify(
```

Comment:
Modify is not supported by sss_override, you would have to combine get/delete/add but it is not worth it. Let's delete it.

Reply:
Same for group.